crates/ide-assists/src/handlers/merge_imports.rs

```diff
@@ -30,14 +33,36 @@ pub(crate) fn merge_imports(acc: &mut Assists, ctx: &AssistContext<'_>) -> Optio
     let (target, edits) = if ctx.has_empty_selection() {
         // Merge a neighbor
         let tree: ast::UseTree = ctx.find_node_at_offset()?;
-        let target = tree.syntax().text_range();
+        let mut target = tree.syntax().text_range();
 
         let edits = if let Some(use_item) = tree.syntax().parent().and_then(ast::Use::cast) {
             let mut neighbor = next_prev().find_map(|dir| neighbor(&use_item, dir)).into_iter();
-            use_item.try_merge_from(&mut neighbor)
+            use_item.try_merge_from(&mut neighbor, &ctx.config.insert_use)
         } else {
             let mut neighbor = next_prev().find_map(|dir| neighbor(&tree, dir)).into_iter();
-            tree.try_merge_from(&mut neighbor)
+            let mut edits = tree.clone().try_merge_from(&mut neighbor, &ctx.config.insert_use);
+
+            if edits.is_none() && ctx.config.insert_use.granularity == ImportGranularity::One {
```

Comment:
```suggestion
            if ctx.config.insert_use.granularity != ImportGranularity::One {
                tree.clone().try_merge_from(&mut neighbor, &ctx.config.insert_use)
            else {
```
This more reasonable to me, since then we don't do work twice.

Reply:
Making this change would break [this test](https://github.com/rust-lang/rust-analyzer/pull/16372/files#diff-697b79caa3c0735c983b89fe0b8e88d6b640e19c929d3bd9019e285ae2774353R290-R298) (and similar ones for `ImportGranularity::One` - the [test case just above it](https://github.com/rust-lang/rust-analyzer/pull/16372/files#diff-697b79caa3c0735c983b89fe0b8e88d6b640e19c929d3bd9019e285ae2774353R281-R289) is the motivation for having this behavior). 

But breaking this behavior (at least for `ImportGranularity::One` only for now) may not be a bad thing because the assist is supposed to "Merges two imports with a common prefix". So may be it's better to provide the functionality equivalent to the test case behavior with a new assist (in a separate PR - something like "normalize import")?

If the above make sense, then I can simplify this implementation even further, while maintaining existing behavior for other import granularities. LMK 🙂

Reply:
I see, then that's probably fine. Could you add a https://docs.rs/cov-mark/latest/cov_mark/ here for one of the corresponding tests? That makes this behavior a bit more  clear